File: src/models/Workout.ts
// src/models/Workout.ts

// 1. Definim enum-ul pentru dificultate exact ca în C#
export enum Difficulty {
    Beginner,      // 0
    Intermediate,  // 1
    Advanced,      // 2
    Pro            // 3
}

// 2. Definim interfața pentru antrenament
export interface Workout {
    id: number;
    name: string;
    description: string;
    difficultyLevel: Difficulty; // Folosește enum-ul definit mai sus
    estimatedDuration: number;   // în minute
    averageCaloriesBurned: number;
}

File: models/Workout.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GymFit.models
{
    public enum Difficulty
    {
        Beginner,
        Intermediate,
        Advanced,
        Pro
    }
    public class Workout
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int CompanyId { get; set; }

        // 🎯 PASUL 10: Legătura directă către obiectul Company pentru Antrenamente
        public Company? Company { get; set; }

        public Difficulty DifficultyLevel { get; set; }
        public int EstimatedDuration { get; set; }
        public int AverageCaloriesBurned { get; set; }

        public Workout()
        {
            Company = null; // Inițializare implicită
        }

        public Workout(int id, string name, string description, int companyId, Difficulty difficultyLevel, int estimatedDuration, int averageCaloriesBurned)
        {
            Id = id;
            Name = name;
            Description = description;
            CompanyId = companyId;
            DifficultyLevel = difficultyLevel;
            EstimatedDuration = estimatedDuration;
            AverageCaloriesBurned = averageCaloriesBurned;
            Company = null; // Inițializare implicită
        }
    }
}